addons/sale/models/sale.py

```diff
@@ -1072,6 +1072,7 @@ def _compute_invoice_status(self):
             elif not float_is_zero(line.qty_to_invoice, precision_digits=precision):
                 line.invoice_status = 'to invoice'
             elif line.state == 'sale' and line.product_id.invoice_policy == 'order' and\
+                    line.product_uom_qty >= 0.0 and\
```

Comment:
The issue is happening because in this case, a negative ordered quantity will always be smaller than the delivered quantity, i.e. the invoice line will end up in the 'upselling' state.

So I'm skipping that state if the ordered quantity is negative.

Reply:
Indeed, considering the delivered qty wouldn't solve the problem :+1: